rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -229,13 +230,16 @@ def _new_sub_callback(self, msg):
         the subscriptions dictionary is updated with the newly incorporated
         subscriptors.
         """
+        # transfer new subscriptions to the "regular" ones
         with self.lock:
-            self.callback(msg, self.new_subscriptions.values())
             self.subscriptions.update(self.new_subscriptions)
             self.new_subscriptions = {}
             self.node_handle.destroy_subscription(self.new_subscriber)
             self.new_subscriber = None
 
+        # send msg to client
+        self.callback(msg)
```

Comment:
This would effectively cause existing clients to receive messages twice when a new client connects. 
I think it would be better to leave it how it was and instead use an [RLock](https://docs.python.org/3/library/threading.html#rlock-objects) to avoid the potential deadlock.